### PR TITLE
Add loglinkconverter plugin

### PR DESCRIPTION
**Pull request by bpowell65536**
When tests are run on a continuous integration system, this plugin converts the local filesystem links to log files given in the test session summary into web links into the artifact store for the job which ran the tests. As a result, the test summary sections for each failing test will contain a clickable link to the log file
for that test, making it easier to quickly diagnose a failing test.

### CODE REVIEW

**Comment by vmalloc** [comment]
Hi! Thanks for the PR!

I think we should consider a better name for this plugin. "Log linker" sounds like it could mean multiple things, and not necessarily what this plugin does. Perhaps "ci_links"? something else?

**Comment by bpowell65536** [comment]
@vmalloc I have made the name change plus a couple of other minor tweaks. I couldn't call the plugin ``ci_links`` because underscores are no longer allowed in plugin names. Do you want me to squash the commits before or after the review?

File: tests/test_cilinks_plugin.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    request.addfinalizer(fin)
+
+    config_override('log.root', request.param)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def deactivate_plugin(request):
+
+    def fin():
```
Comment (vmalloc) [inline tests/test_cilinks_plugin.py]: Please change to @request.addfinalizer

File: tests/test_cilinks_plugin.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    return request.param
+
+
+@pytest.fixture(params=itertools.product(BUILD_URLS, ENV_VAR_NAMES))
+def build_url_env_var(request):
+    url, env_var_name = request.param
+    os.environ[env_var_name] = url
+
+    def fin():
```
Comment (vmalloc) [inline tests/test_cilinks_plugin.py]: Please change to @request.addfinalizer

File: tests/test_cilinks_plugin.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+def log_dir(request, config_override):
+    '''
+    Create a temporary log directory relative to the working directory.
+    This is used instead of the pytest built-in 'tmpdir' because
+    'tmpdir' has an absolute path, and this plugin will not work with
+    absolute log paths.
+    '''
+    os.makedirs(request.param)
+
+    def fin():
```
Comment (vmalloc) [inline tests/test_cilinks_plugin.py]: Please change to @request.addfinalizer

File: tests/test_cilinks_plugin.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    local_log_path = '/'.join(result.get_log_path().split(os.sep))
+    expected_log_link = '%s/artifact/%s' % (plugin.build_url, local_log_path)
+    assert log_link == expected_log_link
+
+
+@pytest.fixture(params=BUILD_URLS)
+def build_url(request):
+    os.environ['BUILD_URL'] = request.param
+
+    def fin():
```
Comment (vmalloc) [inline tests/test_cilinks_plugin.py]: Please change to @request.addfinalizer

**Comment by vmalloc** [comment]
@ayalash your opinion?

**Comment by vmalloc** [comment]
@bpowell65536 one small last request - could you please add a changelog entry for this feature?

**Comment by bpowell65536** [comment]
@vmalloc OK, I'll add the changelog entry now.

@ayalash Thanks for the information about plugin names. I tried using dashes in the plugin name initially but they are no longer allowed. I'll change the plugin name too.

I put detailed usage and configuration information for the plugin in its docstring, but when I look at the other built-in plugins they put their detailed usage information in the documentation's "Built-in plugins" page. Do you want me to add a section for this plugin there?

**Comment by vmalloc** [comment]
@bpowell65536 not critical, but it could be nice to have all the built-in plugins documented there.

**Comment by vmalloc** [comment]
Many thanks for your contribution and patience!

**Comment by bpowell65536** [comment]
@vmalloc No worries! I haven't squashed the commits on this branch yet - did you do that?

**Comment by vmalloc** [comment]
No, forgot to. No biggie though - there is a merge commit to mark the merge so I'm not worried.

**Pull request merged**